opengovsg/FormSG
# feat: email notifications for mrf completed workflows

## Problem

Admins want to be notified when an mrf is completed.   
Closes FRM-1823

## Solution

Implement MRF v2 email notifications

**Breaking Changes** 

No - this PR is backwards compatible  

**Features**:
Send completion email to selected emails and emails assigned to steps when mrf workflow is completion. 

**Improvements**:
**Bug Fixes and UI improvements**:
| Description | Before | After | 
| ----------- | ------ | ----- |
| on small screens, the mrf text overflows, it should wrap to a new line. There should be a maxW on the `LogicBadge` component | <img height="200" alt="image" src="https://github.com/user-attachments/assets/21436633-37cf-420b-a1e5-ca5476714ce2"> | <img height="200" alt="image" src="https://github.com/user-attachments/assets/2becf698-03cb-4d0c-8c15-94ebf18f38ca"> |
| tooltip popups should not block the component it is describing for. Hence, I have configured the popup placement with `useBreakpointValue` to prevent obfuscating the mrf fields. | <img width="715" alt="image" src="https://github.com/user-attachments/assets/ece3e7c2-2a9d-4690-b609-25a782d9a2d8"> | Moved the tooltip up after consulting with Alicia |

## Before & After Screenshots

| Description | Before | After | 
| ----------- | ------ | ----- |
| Email notifications page | | <img height="300" alt="image" src="https://github.com/user-attachments/assets/ae3d0a88-2ddf-4bb6-a949-4ac0d67e3cab"> |
| MRF builder | | <img height="300" alt="image" src="https://github.com/user-attachments/assets/bd9d3d43-17a4-43d6-9059-27f09ec3be1b"> | 
| Email sent | | <img height="300" alt="image" src="https://github.com/user-attachments/assets/7f661793-44a7-4cb5-a9df-7ef366de184f"> |  

## Manual Tests


TC1: Verify that mrf still works (regression test)   
- [ ] Create mrf with 3 steps, assign step 1 to email field 1, step 2 to own email, step 3 to email field 2 
- [ ] Complete the workflow, ensure that the emails are still sent and all the data is filled correctly

TC2: Verify that mrf sends email workflow notifications as per steps.
- [ ] Use same mrf from TC1, add email field 1 as respondent to Step 1 of mrf
- [ ] Go to email notifications, select steps 1 and 2 for email notifications 
- [ ] Complete the workflow, ensure that can complete & email field 1 email + own email in step 2 gets the workflow completion email 
- [ ] Verify step 3 email does not get an email. 

TC3: Verify that mrf sends email workflow notifications as per steps & static emails 
- [ ] From TC2, add 2 static emails to email notifications page. Remove step 1 and 2 from email and add step 3.  
- [ ] Complete workflow. 
- [ ] Verify 2 static email + step 3 respondent email gets email completion email.
